### PR TITLE
First Draft for a guide on how to contribute to pysyft.

### PR DESCRIPTION
**Pull request by Ogofo**
This was created by me and @Nivek92. We propose this as the first draft of an entry point for new contributors and defining rules for current contributors. This way we can formalize the rules for contributions and aim for a clean, readable and well-designed code base.

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+If you want to work on an open issue, please post a comment telling that you will work on that issue, we will assign you ass the assignee then.
+
+**Caution**: We try our best to keep the assignee up-to-date but as we are all humans with our own schedule delays are possible, so make sure to check the comments once before you start working on an issue even when noone is assigned to it.
+
+
+## Writing Test Cases
+Always make sure to create the necessary tests and keep test coverage at 100%. You can always ask for help in slack or via github if you don't feel confidant about your tests.
+
+We aim to have a 100% test coverage, and the Travis CI will fail if the coverage is below this value. You can evaluate your coverage using the following commands.
```
Comment (jvmancuso) [inline CONTRIBUTORS.md]: how was the 100% test coverage chosen?  from what I've heard, it's not a great measure of test suite quality/robustness (a la Goodhart's law), and as such many projects go with something in the 70-90% range

Comment (robert-wagner) [reply CONTRIBUTORS.md]: The idea behind 100% test coverage is that we want to be intentional about what we are not testing. We could use pragma: no cover for some functions if we feel they don't need a test or can't be tested but that should be a well thought out choice

Comment (robert-wagner) [reply CONTRIBUTORS.md]: The idea behind 100% test coverage is that we want to be intentional about what we are not testing. We could use pragma: no cover for some functions if we feel they don't need a test or can't be tested but that should be a well thought out choice

File: CONTRIBUTORS.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+You documentation should not describe the obvious, but explain what's the intention behind the code and how you tried to realize your intention.
+
+You should also document non self-explanatory code fragments e.g. complicated for-loops. Again please do not just describe what each line is doing but also explain the idea behind the code fragment and why you decided to use that exact solution.
+
+#### Imports
+For better merge compatibility each import is within a separate line. Mulitple imports from one package are written in one line each.
+
+Example:
+```
```
Comment (robert-wagner) [inline CONTRIBUTORS.md]: Maybe also put something about relative imports

Comment (Ogofo) [reply CONTRIBUTORS.md]: Sure! Good idea, do you have a suggestion what should be added?

File: CONTRIBUTORS.md (hunk 1 above)
Comment (LaRiffle) [reply CONTRIBUTORS.md]: 100% is not possible nor practical, but we can start with a high barrier like 90% - 95% and lower than if needed?

**Pull request merged**